### PR TITLE
Add debug mode to is_has_traits_almost_equal.

### PR DESCRIPTION
**Pull request by jonathanrocher**
- Add new `debug` keyword argument. Only useful when an exception is raised by the code, to understand what triggered it. In normal usage, the equality failure will also report what attribute path failed equality.
- General code improvements.

### CODE REVIEW

File: app_common/traits/assertion_utils.py [hunk 1]
```diff
@@ -7,61 +7,40 @@
...
-        one of its children attribute name.
-
-    check_type : bool
-        Check whether the 2 objects have the same type? True by default.
     """
-    is_equal, msg = is_has_traits_almost_equal(
-        obj1, obj2, eps=eps, ignore=ignore, check_type=check_type
-    )
+    is_equal, msg = is_has_traits_almost_equal(obj1, obj2, **kwargs)
     error_msg = "HasTraits objects are not almost equal: {}".format(msg)
```
Comment (chamshop) [inline app_common/traits/assertion_utils.py:39]: Should this line be changed to say they *are* almost equal?

File: app_common/traits/assertion_utils.py [hunk 2]
```diff
@@ -96,16 +67,8 @@ def assert_values_not_almost_equal(val1, val2, eps=1e-9, ignore=()):
...
-    ignore : sequence of str
-        List of attribute names to skip from comparisons of objects. These are
-        passed recursively to elements of the values being compared and their
-        children.
+    kwargs : dict
+        Additional keyword arguments passed to `is_val_almost_equal`.
     """
-    is_equal, msg = is_val_almost_equal(val1, val2, eps=eps, ignore=ignore)
+    is_equal, msg = is_val_almost_equal(val1, val2, **kwargs)
     assert_false(is_equal, "Values are not almost equal: {}".format(msg))
```
Comment (chamshop) [inline app_common/traits/assertion_utils.py:74]: Should this line be changed to say the values *are* almost equal?

**Review by chamshop — APPROVED** [review]
LGTM, just a couple potential documentation fixes.

File: app_common/traits/assertion_utils.py (hunk 1 above)
Comment (jonathanrocher) [reply app_common/traits/assertion_utils.py:39]: Correct!

**Comment by jonathanrocher** [comment]
Thanks for the review. I corrected the message mistakes.

**Comment by jonathanrocher** [comment]
Test failure is unrelated. Merging...

**Pull request merged**